internal/provider/resource_provider_integration_authorization.go

```diff
@@ -0,0 +1,399 @@
+		customerTenantId := azureMap[paramAzureCustomerTenantId].(string)
+
+		azureIntegrationConfig := &piv2.PimV2AzureIntegrationConfig{
+			Kind:                  "AzureIntegrationConfig",
+			CustomerAzureTenantId: &customerTenantId,
+		}
+		updateConfig = piv2.PimV2AzureIntegrationConfigAsPimV2IntegrationUpdateConfigOneOf(azureIntegrationConfig)
+
+	case providerGcp:
+		gcpConfig := d.Get(paramGcpAuth).([]interface{})
+		if len(gcpConfig) == 0 {
+			return diag.Errorf("GCP configuration is required for provider %s", provider)
+		}
+		gcpMap := gcpConfig[0].(map[string]interface{})
+		customerServiceAccount := gcpMap[paramGcpCustomerServiceAccount].(string)
+
+		gcpIntegrationConfig := &piv2.PimV2GcpIntegrationConfig{
+			Kind:                         "GcpIntegrationConfig",
+			CustomerGoogleServiceAccount: &customerServiceAccount,
+		}
+		updateConfig = piv2.PimV2GcpIntegrationConfigAsPimV2IntegrationUpdateConfigOneOf(gcpIntegrationConfig)
+
+	default:
+		return diag.Errorf("Unsupported provider: %s", provider)
+	}
+
+	var updatedIntegration *piv2.PimV2Integration
+
+	// Only PATCH if integration is still in DRAFT status
+	if status == "DRAFT" {
+		// PATCH the integration with the customer configuration to change status from DRAFT to CREATED
+		tflog.Info(ctx, fmt.Sprintf("Updating provider integration %q with customer configuration", integrationId))
+
+		updateReq := piv2.PimV2IntegrationUpdate{}
+		updateReq.SetConfig(updateConfig)
+		updateReq.SetEnvironment(piv2.ObjectReference{Id: environmentId})
+
+		patchApiReq := c.piV2Client.IntegrationsPimV2Api.UpdatePimV2Integration(c.piV2ApiContext(ctx), integrationId).PimV2IntegrationUpdate(updateReq)
+		patchedIntegration, _, err := patchApiReq.Execute()
+		if err != nil {
+			return diag.Errorf("Failed to update provider integration %q with customer configuration: %s", integrationId, createDescriptiveError(err))
+		}
+		updatedIntegration = &patchedIntegration
+		tflog.Info(ctx, fmt.Sprintf("Successfully updated provider integration %q. Status: %s", integrationId, updatedIntegration.GetStatus()))
+	} else {
+		// Integration is already CREATED, skip PATCH and use existing integration
+		tflog.Info(ctx, fmt.Sprintf("Integration %q is already in CREATED status, skipping configuration update", integrationId))
+		updatedIntegration = &integration
+	}
+
+	// Set the resource ID and populate data first (so we get setup information even if validation fails)
+	d.SetId(integrationId)
+
+	// Read the integration data to populate outputs (including multi-tenant app ID)
+	readDiags := providerIntegrationAuthorizationRead(ctx, d, meta)
+	if readDiags.HasError() {
+		return readDiags
+	}
+
+	// Always validate the integration
+	return validateIntegrationSetup(ctx, c, integrationId, environmentId, status, updateConfig)
+}
+
+func providerIntegrationAuthorizationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if d.Id() == "" {
+		return nil
+	}
```

Comment:
Could you add a comment here how it works? I'm not sure we're using it this check implementing other resources.